src/main/java/org/codehaus/mojo/flatten/DirectDependenciesInheritanceAssembler.java

```diff
@@ -0,0 +1,158 @@
+  * to you under the Apache License, Version 2.0 (the
+  * "License"); you may not use this file except in compliance
+  * with the License.  You may obtain a copy of the License at
+  *
+  *   http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing,
+  * software distributed under the License is distributed on an
+  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  * KIND, either express or implied.  See the License for the
+  * specific language governing permissions and limitations
+  * under the License.
+  */
+package org.codehaus.mojo.flatten;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.enterprise.inject.Specializes;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.maven.model.Model;
+import org.apache.maven.model.ModelBase;
+import org.apache.maven.model.building.ModelBuildingRequest;
+import org.apache.maven.model.building.ModelProblemCollector;
+import org.apache.maven.model.inheritance.DefaultInheritanceAssembler;
+import org.apache.maven.model.merge.MavenModelMerger;
+
+/**
+ * The DefaultInheritanceAssembler is encapsulating the InheritanceModelMerger.
+ * The only way to override functionality needed is to define an own InheritanceAssembler 
+ * to provide the needed ModelMerger.
+ * The container is providing the InheritanceAssembler.
+ * This class needs to be configured in the 
+ * src/main/resources/META-INF/sisu/javax.inject.Named to be loaded by the container.
+ *
+ * see issue https://github.com/mojohaus/flatten-maven-plugin/issues/220
+ *
+ * @author kemalsoysal
+ * @author ralfluebeck
+ *
+ */
+@Specializes
+@Singleton
+@Named
+public class DirectDependenciesInheritanceAssembler extends DefaultInheritanceAssembler {
+
+	protected InheritanceModelMerger merger = new DirectDependenciesInheritanceModelMerger();
+
+	/**
+	 * copied from super implementation because it is private
+	 */
+	private static final String CHILD_DIRECTORY = "child-directory";
+
+	/**
+	 * copied from super implementation because it is private
+	 */
+	private static final String CHILD_DIRECTORY_PROPERTY = "project.directory";
+
+	protected FlattenDependencyMode flattenDependencyMode;
+	
+	/**
+	 * 
+	 */
+	public DirectDependenciesInheritanceAssembler() {
+	}
+
+	@Override
+	public void assembleModelInheritance(Model child, Model parent, ModelBuildingRequest request,
+			ModelProblemCollector problems) {
+		Map<Object, Object> hints = new HashMap<>();
+		String childPath = child.getProperties().getProperty(CHILD_DIRECTORY_PROPERTY, child.getArtifactId());
+		hints.put(CHILD_DIRECTORY, childPath);
+		hints.put(MavenModelMerger.CHILD_PATH_ADJUSTMENT, getChildPathAdjustment(child, parent, childPath));
+		merger.merge(child, parent, false, hints);
+	}
+
+	/**
+	 * copied from super implementation because it is private though the adjustment
+	 * is only for compatibility due to the comment with Maven 2.0
```

Comment:
Maven 2.x is not supported by plugin.
Minimum is 3.2.5

Reply:
I have no idea, what really needs to be done here. I have just copied the private method to be able to make additional implementation to the base class.
The base class should be refactored to allow proper overriding/implementation of the functionality.
But the base class is not in the scope of this plugin.
Hence the base class implementation must be observed to reflect needed changes in this specific InheritanceAssembler. Maybe we should add a TODO to the method implementation?